generators/client/templates/react/src/main/webapp/app/modules/administration/metrics/metrics-modal.tsx.ejs

```diff
@@ -1,57 +1,161 @@
+          threadDumpRunnable++;
+          break;
+        case 'WAITING':
+          threadDumpWaiting++;
+          break;
+        case 'TIMED_WAITING':
+          threadDumpTimedWaiting++;
+          break;
+        case 'BLOCKED':
+          threadDumpBlocked++;
+          break;
+        default:
+          break;
+      }
+    });
+
+    threadDumpAll = threadDumpRunnable + threadDumpWaiting + threadDumpTimedWaiting + threadDumpBlocked;
+    return { threadDumpAll, threadDumpRunnable, threadDumpWaiting, threadDumpTimedWaiting, threadDumpBlocked };
+  };
+
+  getBadgeClass = threadState => {
+    if (threadState === 'RUNNABLE') {
+      return 'badge-success';
+    } else if (threadState === 'WAITING') {
+      return 'badge-info';
+    } else if (threadState === 'TIMED_WAITING') {
+      return 'badge-warning';
+    } else if (threadState === 'BLOCKED') {
+      return 'badge-danger';
+    }
+  };
+
+  updateBadgeFilter = badge => () => {
+    this.setState({ badgeFilter: badge });
+  };
+
+  updateSearchFilter = () => event => {
+    this.setState({ searchFilter: event.target.value });
+  };
+
+  render() {
+    const { showModal, handleClose, threadDump } = this.props;
+    let counters = null;
+    let filteredList = null;
+    if (threadDump && threadDump.threads) {
+      counters = this.computeCounters();
+      filteredList = this.computeFilteredList();
+    }
+
+    return <Modal
+      isOpen={showModal} modalTransition={{ timeout: 20 }} backdropTransition={{ timeout: 10 }}
+      toggle={handleClose} className="modal-lg">
+      <ModalHeader toggle={handleClose}>Threads dump</ModalHeader>
+      <ModalBody>
+        <Badge color="primary" onClick={this.updateBadgeFilter('')}>All&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpAll : null}</span>
+        </Badge>&nbsp;
+        <Badge color="success" onClick={this.updateBadgeFilter('RUNNABLE')}>Runnable&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpRunnable : null}</span>
+        </Badge>&nbsp;
+        <Badge color="info" onClick={this.updateBadgeFilter('WAITING')}>Waiting&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpWaiting : null}</span>
+        </Badge>&nbsp;
+        <Badge color="warning" onClick={this.updateBadgeFilter('TIMED_WAITING')}>Timed Waiting&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpTimedWaiting : null}</span>
+        </Badge>&nbsp;
+        <Badge color="danger" onClick={this.updateBadgeFilter('BLOCKED')}>Blocked&nbsp;
+          <span className="badge badge-pill badge-default">{counters ? counters.threadDumpBlocked : null}</span>
+        </Badge>&nbsp;
+        <div className="mt-2">&nbsp;</div>
+        <Input type="text" className="form-control" placeholder="Filter by Lock Name..."
+               onChange={this.updateSearchFilter()}/>
```

Comment:
no need for extra closure here as we are not binding any value, change the method also to just
`updateSearchFilter  = event => this.setState(...);`